packages/camera/camera/example/ios/RunnerTests/CameraMethodChannelTests.m

```diff
@@ -0,0 +1,101 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import camera;
+@import XCTest;
+@import AVFoundation;
+#import <OCMock/OCMock.h>
+
+@interface FLTThreadSafeFlutterResult ()
+@property(readonly, nonatomic) FlutterResult flutterResult;
+@end
+
+@interface MockFLTThreadSafeFlutterResult : FLTThreadSafeFlutterResult
+@property(readonly, nonatomic) NSNotificationCenter *notificationCenter;
+@property(nonatomic, nullable) id receivedResult;
+@end
+
+@implementation MockFLTThreadSafeFlutterResult
+/**
+ Initialize with a notification center.
+ */
+- (id)initWithNotificationCenter:(NSNotificationCenter *)notificationCenter {
+  self = [super init];
+  _notificationCenter = notificationCenter;
```

Comment:
I'm confused about the use of the notification center. Why not just have the mock have a public `expectation` property that it fulfills when any result is called, and that tests can wait on?

Reply:
You are right, the notification center is not needed.